### ISSUE TITLE
NAs break `getTopTaxa` 

### ISSUE THREAD

**Issue by microsud** [microbiome/mia#158]
During testing bumped on this bug. It seems that after transformation some NA is introduced? Which then silently kills the `getTopTaxa` 
```
library(microbiomeDataSets)
tse <- artificialgut()
tse <- transformCounts(tse, method = "relabundance")
#taxa <- getTopTaxa(silverman, 2)
# WORKS
getTopTaxa(tse,
           method = "mean",
           top = 5,
           abund_values = "counts")
[1] "seq_1" "seq_2" "seq_3" "seq_4" "seq_5"

# Fails without warning
getTopTaxa(tse,
           method = "mean",
           top = 5,
           abund_values = "relabundance")
#Output
character(0)
```
We have this at the moment in `getTopTaxa`
line 104-107
```
taxs <- switch(method,
                       mean = rowMeans2(assay(x, abund_values)),
                       sum = rowSums2(assay(x, abund_values)),
                       median = rowMedians(assay(x, abund_values)))

``` 
Should we include `na.rm = TRUE` to these funs?

**Comment by antagomir** [comment]
Adding `na.rm` as an option (default `TRUE`?) would be good imo.

But why there are NAs in relabundances but not in counts? Doesnẗ that indicate other problems?

**Comment by microsud** [comment]
Yes just checked and there are some samples with zero counts in `artificialgut`

```
tse <- artificialgut()
mia::summary(tse)
```
>Error: There are samples that sum to 1 or less counts. Try to supply raw counts 

This is why there is an issue.  
However, we still need to have `na.rm` possibly to avoid such issues ? 

EDIT: Also is it okay to modify the `artificialgut` data? This is the original data from publication and I think we need to keep it as it? But we can use this as a case for some sections in `OMA` to show how we remove samples that sum to zero?

EDIT- Should `transformCounts` return a warning in such cases?

**Comment by antagomir** [comment]
Right - 5 samples with 0 counts and 2 with 1 count.. (!?) and then a few with very low counts.

I do not think it is OK to modify the original data. Although: it could be justified to remove samples with low read counts. But I think it is more transparent to keep them in this data set and do that as a preprocessing step after data retrieval, when necessary. Potentially useful for some OMA examples, yes.

I would say that `transformCounts` does not have to return a warning when there are more than 0 reads; BUT when there are 0 counts it should return 0 relative abundances instead of NAs and a warning might be justified.

**Comment by FelixErnst** [comment]
> BUT when there are 0 counts it should return 0 relative abundances instead of NAs and a warning might be justified.

I would say this is a transformation of data and not right way to go. `x/0` is `NA` in R and it will always be that. Wrapping the user in bubblewrap doesn't help in the end.

**Comment by antagomir** [comment]
Yes, in a sense.

But if there abundance is 0 in counts, it would be in a sense correct to say that also relative abundance is 0.

But it is also incorrect in certain ways for the reason of x/0 being NA. 

I am ok with the latter solution as well, it can be more informative and failure can be justified in those cases. The main issue is that this may escape attention as had happened in the reported case above. Hence, I would support adding warning to the transformations in this kind of cases.

**Comment by microsud** [comment]
I can add an argument for na.rm with default TRUE to this function. Will create a new PR.

**Comment by antagomir** [comment]
Yep. But `na.rm=TRUE` will not help with `x/0` I guess.

**Comment by microsud** [comment]
For `getTopTaxa` it should be okay and user can check by changing default. We can keep `transformCounts` as is now and see later if there is urgent/valid reason to address this.

**Comment by FelixErnst** [comment]
> But if there abundance is 0 in counts, it would be in a sense correct to say that also relative abundance is 0.

Nope. that is information not available, therefore NA. 

The question you ask with that function is `What is the relative abundance?` If you would ask `Is there abundance?` or `What is the abundance`, you could return a non-NA value, but with the key word "relative", IMO, the only answer can be NA.

@microsud Thanks for the PR

**Comment by antagomir** [comment]
I certainly agree that in one sense the answer is NA and that's ok.

**Comment by FelixErnst** [comment]
Any news on this?

**Comment by microsud** [comment]
PR for this #199

**Comment by antagomir** [comment]
Closed via #205
